cmd/loadTests.go

```diff
@@ -53,6 +55,102 @@ var (
 	threadsWG                          sync.WaitGroup
 )
 
+type LogData struct {
+	Timestamp                         string      `json:"timestamp"`
+	MachineName                       string      `json:"machineName"`
+	BinaryDetails                     string      `json:"binaryDetails"`
+	NumberOfThreads                   int         `json:"Number of threads"`
+	NumberOfUsersPerThread            int         `json:"Number of users per thread"`
+	BatchSize                         int         `json:"Batch size per thread"`
+	NumberOfUsers                     int         `json:"Total number of users"`
+	LoadTestCompletionStatus          string      `json:"loadTestCompletionStatus"`
+	AverageTimeToSpinUpUsers          FloatFormat `json:"Average Time taken to spin up users (sec)"`
+	AverageTimeToCreateResources      FloatFormat `json:"Average Time taken to Create Resources (sec)"`
+	AverageTimeToRunPipelines         FloatFormat `json:"Average Time taken to Run Pipelines (sec)"`
+	UserCreationFailureCount          int64       `json:"Number of times user creation failed"`
+	UserCreationFailurePercentage     FloatFormat `json:"User creation failed percentage"`
+	ResourceCreationFailureCount      int64       `json:"Number of times resource creation failed"`
+	ResourceCreationFailurePercentage FloatFormat `json:"Resource creation failed percentage"`
+	PipelineRunFailureCount           int64       `json:"Number of times pipeline run failed"`
+	PipelineRunFailurePercentage      FloatFormat `json:"Pipeline run failed percentage"`
+}
+
+// to marshall json float as .2f% we need the MarshalJSON function
+
+type FloatFormat float64
+
+func (f FloatFormat) MarshalJSON() ([]byte, error) {
```

Comment:
Let's not limit precision of numbers we are pushing to the JSON.

Reply:
+1

Reply:
Sure, fixed